### PR TITLE
Add more functionality to Keithley 2182A device driver

### PR DESCRIPTION
**Pull request by therealpyro**
These commits add more functionality to the sense command layer and introduce the trace and system command layer.

### CODE REVIEW

**Comment by therealpyro** [comment]
I don't know why Travis fails to build. When I test it locally, everything works fine.

**Comment by p3trus** [comment]
Hi. Thank you for your contribution. I will look at it in the next few days.

File: slave/keithley/k2182.py [hunk 1]
```diff
@@ -69,16 +69,106 @@ class Sense(Driver):
...
+        self.points = Command(
+            ':TRAC:POIN?',
+            ':TRAC:POIN',
+            Integer(min=2, max=1024)
+        )
+        self.feed = Command(
+            ':TRAC:FEED?',
+            ':TRAC:FEED',
+            Mapping({'sense': 'SENS', 'calculate': 'CALC', None: 'NONE'})
+        }
```
Comment (p3trus) [inline slave/keithley/k2182.py]: The Problem seems to be this typo.

**Comment by therealpyro** [comment]
I fixed two typos in the code. The Travis build now succeeds.

File: slave/keithley/k2182.py [hunk 2]
```diff
@@ -69,16 +69,106 @@ class Sense(Driver):
...
+            Mapping({'next': 'NEXT', 'never': 'NEV'})
+        )
+
+    def clear(self):
+        """Clear readings from buffer."""
+        self._write(':TRAC:CLEAR')
+
+    def free(self):
+        """Query bytes available and bytes in use."""
+        response = self._query((':TRAC:FREE?', String))
```
Comment (p3trus) [inline slave/keithley/k2182.py]: This should work as well:

    return response = self._query((':TRAC:FREE?', [Float, Float]))

**Comment by p3trus** [comment]
@therealpyro Merged your contribution and updated the code a bit. You might want to check if it's still working.

**Pull request merged**